gossipd/test/run-extended-info.c

```diff
@@ -0,0 +1,790 @@
+	"    },\n"
+	"    \"extensions\" : [ {\n"
+	"      \"encoding\" : \"COMPRESSED_ZLIB\",\n"
+	"      \"array\" : [ 1, 2, 4 ]\n"
+	"    } ]\n"
+	"  },\n"
+	"  \"hex\" : \"01050f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e22060019000000000000002fc80000000000003cc4000000000045a6c4010c01789c6364620100000e0008\"\n"
+	"}\n",
+	"{\n"
+	"  \"msg\" : {\n"
+	"    \"type\" : \"QueryShortChannelIds\",\n"
+	"    \"chainHash\" : \"0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206\",\n"
+	"    \"shortChannelIds\" : {\n"
+	"      \"encoding\" : \"COMPRESSED_ZLIB\",\n"
+	"      \"array\" : [ \"0x0x14200\", \"0x0x46645\", \"0x69x42692\" ]\n"
+	"    },\n"
+	"    \"extensions\" : [ {\n"
+	"      \"encoding\" : \"COMPRESSED_ZLIB\",\n"
+	"      \"array\" : [ 1, 2, 4 ]\n"
+	"    } ]\n"
+	"  },\n"
+	"  \"hex\" : \"01050f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206001801789c63600001f30a30c5b0cd144cb92e3b020017c6034a010c01789c6364620100000e0008\"\n"
+	"}\n",
+};
+
+static void get_chainhash(const char *test_vector,
+			  const jsmntok_t *obj,
+			  struct bitcoin_blkid *chain_hash)
+{
+	const jsmntok_t *tok = json_get_member(test_vector, obj, "chainHash");
+	size_t hexlen = tok->end - tok->start;
+	assert(hex_decode(test_vector + tok->start, hexlen,
+			  chain_hash, sizeof(*chain_hash)));
+}
+
+static u8 *get_scid_array(const tal_t *ctx,
+			  const char *test_vector,
+			  const jsmntok_t *obj)
+{
+	const jsmntok_t *scids, *arr, *encoding, *t;
+	size_t i;
+	u8 *encoded;
+
+	scids = json_get_member(test_vector, obj, "shortChannelIds");
+	arr = json_get_member(test_vector, scids, "array");
+
+	encoded = encoding_start(ctx);
+	encoding = json_get_member(test_vector, scids, "encoding");
+	json_for_each_arr(i, t, arr) {
+		struct short_channel_id scid;
+		assert(json_to_short_channel_id(test_vector, t, &scid, false));
+		encoding_add_short_channel_id(&encoded, &scid);
+	}
+	if (json_tok_streq(test_vector, encoding, "UNCOMPRESSED")) {
+		encoding_end_no_compress(&encoded, 1);
+		encoded[0] = SHORTIDS_UNCOMPRESSED;
+	} else {
+		assert(json_tok_streq(test_vector, encoding, "COMPRESSED_ZLIB"));
+		assert(encoding_end_zlib(&encoded, 1));
+		encoded[0] = SHORTIDS_ZLIB;
+	}
+
+	return encoded;
+}
+
+static u8 *test_query_channel_range(const char *test_vector, const jsmntok_t *obj)
+{
+	struct bitcoin_blkid chain_hash;
+	u32 firstBlockNum, numberOfBlocks;
+	const jsmntok_t *opt, *t;
+	struct tlv_query_channel_range_tlvs *tlvs
+		= tlv_query_channel_range_tlvs_new(NULL);
+	u8 *msg;
+	size_t i;
+
+	get_chainhash(test_vector, obj, &chain_hash);
+	assert(json_to_number(test_vector, json_get_member(test_vector, obj, "firstBlockNum"), &firstBlockNum));
+	assert(json_to_number(test_vector, json_get_member(test_vector, obj, "numberOfBlocks"), &numberOfBlocks));
+	opt = json_get_member(test_vector, obj, "extensions");
+	json_for_each_arr(i, t, opt) {
+		assert(json_tok_streq(test_vector, t,
+				      "WANT_TIMESTAMPS | WANT_CHECKSUMS"));
```

Comment:
would it be worthwhile to also test these separately?

Reply:
Yes, but the test vectors are grossly insufficient.  We need behavoural tests